pkg/controllers/helpers.go

```diff
@@ -0,0 +1,24 @@
+package controllers
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func p[T any](v T) *T {
+	return &v
+}
+
+func setOwnerRef(ownee, owner client.Object) {
```

Comment:
It's unclear that this is passing by pointer the `ownee` object, so the mutation seems to die on the scope. I'm guessing that's not the actual behaviour, or this wouldn't do anything at all.

Nit: I usually prefer to return the ownee, as that makes it clear that we are modifying it, but otherwise, an indication of the mutation would be useful here.

Reply:
I get your point. Although, given that we are using the `SetOwnerReferences`  setter, the pointer management should be managed by the `SetOwnerReferences` implementation